sbncode/FlashMatch/FlashPredict.hh

```diff
@@ -148,87 +292,74 @@ private:
   const double fBeamWindowStart, fBeamWindowEnd;
   const double fFlashStart, fFlashEnd;
   const unsigned fTimeBins;
-  const bool fSelectNeutrino, fOnlyPrimaries;
+  const bool fSelectNeutrino;
+  const bool fOnlyCollectionWires;
+  const bool fForceConcurrence;
   const bool fUseUncoatedPMT, fUseOppVolMetric;//, fUseCalo;
   const bool fUseARAPUCAS;
+  const bool fStoreCheatMCT0;
   const std::string fInputFilename;
   const bool fNoAvailableMetrics, fMakeTree;
   const double fChargeToNPhotonsShower, fChargeToNPhotonsTrack;
-  const double fMinHitQ, fMinSliceQ, fQScale;
-  const double fMinOpHPE, fMinFlashPE, fPEScale;
-  std::string fDetector; // SBND or ICARUS
-  bool fSBND, fICARUS;
-  std::unique_ptr<opdet::PDMapAlg> fPDMapAlgPtr;
+  const double fMinHitQ, fMinSpacePointQ, fMinParticleQ, fMinSliceQ;
+  const unsigned fMaxFlashes;
+  const double fMinOpHPE, fMinFlashPE;
+  const art::ServiceHandle<geo::Geometry> fGeometry;
+  const std::string fDetector; // SBND or ICARUS
+  const bool fSBND, fICARUS;
+  const std::unique_ptr<opdet::PDMapAlg> fPDMapAlgPtr;
+  const size_t fNTPC;
   const int fCryostat;  // =0 or =1 to match ICARUS reco chain selection
   // geo::CryostatID fCryostat;  // TODO: use this type instead
-  std::unique_ptr<geo::CryostatGeo> fGeoCryo;
-  const int fNBins;
+  const std::unique_ptr<geo::CryostatGeo> fGeoCryo;
+  const std::list<double> fWiresX_gl;
   const double fDriftDistance;
-  size_t fNTPC;
+  const int fXBins;
+  const double fXBinWidth;
+  const std::string fRR_TF1_fit, fRatio_TF1_fit;
+  unsigned fYBins,fZBins;
+  double fYLow, fYHigh, fZLow, fZHigh;
+  double fYSkewThreshold, fZSkewThreshold;
+  const double fYBiasSlope, fZBiasSlope;
   unsigned fDriftVolumes;
   unsigned fTPCPerDriftVolume;
   const unsigned fOpDetNormalizer;
   const double fTermThreshold;
-  std::list<double> fWiresX_gl;
-  // std::vector<double> fPMTChannelCorrection;
 
-  unsigned fPeakCounter = 0;
-  std::vector<recob::OpHit>::iterator fOpH_beg, fOpH_end;
-  const art::ServiceHandle<geo::Geometry> geometry;
+  const unsigned kRght = 0;
+  const unsigned kLeft = 1;
+
+  const unsigned kActivityInRght = 100;
+  const unsigned kActivityInLeft = 200;
+  const unsigned kActivityInBoth = 300;
 
   // root stuff
   TTree* _flashmatch_nuslice_tree;
-  std::unique_ptr<TH1D> fOpHitsTimeHist;
-  // double rrMax, peMax;
-  // TSpline3 rr_m_InvSpl, rr_h_InvSpl, rr_l_InvSpl;
-  // TSpline3 pe_m_InvSpl, pe_h_InvSpl, pe_l_InvSpl;
   struct Fits {
     double min, max;
     std::unique_ptr<TF1> f;
   };
-  std::array<Fits, 3> rrFits;
-  std::array<Fits, 3> peFits;
-  const std::array<std::string, 3> suffixes{"l", "h", "m"};
-  // const std::string kPolFit = "pol3";
+  TH2D* fRRH2; TH2D* fRatioH2;
+  const unsigned kMinEntriesInProjection = 100;
+  std::array<Fits, 3> fRRFits;
+  std::array<Fits, 3> fRatioFits;
+  const std::array<std::string, 3> kSuffixes{"l", "h", "m"};// low, high, medium
```

Comment:
If these are known at compile time, is there a reason not to use a `constexpr`

Reply:
Turns out C++ has a thing against making std::string constexpr. Whilst there are ways around it, there's really no point. I've stopped using the method that called for `kSuffixes`.